perceval/runtime/remote_processor.py

```diff
@@ -200,3 +201,62 @@ def source(self):
     def source(self, source: Source):
         # TODO: Implement source setter, setting parameters to be sent remotely
         raise NotImplementedError("Source setting not implemented for remote processors")
+
+    def _compute_sample_of_interest_probability(self,
+                                                transmittance: float = 0.06,
+                                                param_values: dict = None
+                                                ) -> float:
+        # TODO retrieve transmittance from the cloud worker
+        losses = 1 - transmittance
+        n = self._input_state.n
+        photon_filter = n
+        if self._min_detected_photons is not None:
+            photon_filter = self._min_detected_photons
+            if photon_filter > n:
+                return 0
+        if photon_filter < 2:
+            return 1
+
+        # Simulation with a noisy source (only losses)
+        c = self.linear_circuit(flatten=True).copy()
+        if param_values:
+            for n, v in param_values.items():
+                c.param(n).set_value(v)
+        lp = Processor(SLOSBackend(), c, Source(losses=losses))
```

Comment:
Can't you use instead str "SLOS" instead of "SLOSBackend()" to avoid "from perceval.backends import SLOSBackend"? I know it's a private class and everything but remote_processor should not depend on backends. Moreover isn't it better to pass it as a parameter (that can be SLOS by default) that can be change by the user ?

Reply:
Ok, so this low level dependency is hidden. However, `RemoteProcessor` does depend on SLOS back-end now we simulate circuits in order to estimate the shots/samples conversion ratio.